### PR TITLE
VpnTunnel - show global and individual connection status

### PR DESCRIPTION
**Pull request by mdelaossa**
## When auto == "route"
image

## When auto has another value
image

### CODE REVIEW

File: src/deploy/vpn-tunnel/index.ts [hunk 1]
```diff
@@ -72,9 +85,24 @@ export const deserializeDeployVpnTunnel = (
     createdAt: payload.created_at,
     updatedAt: payload.updated_at,
     stackId: extractIdFromLink(payload._links.stack),
+    state: payload.state as DeployVpnTunnelState,
+    tunnelAttributes: camelizeKeys(payload.tunnel_attributes),
   };
 };
 
+const camelizeKeys: any = (obj: any) => {
```
Comment (neurosnap) [inline src/deploy/vpn-tunnel/index.ts:93]: Can we move this into `string-utils`?

File: src/schema/factory.ts [hunk 2]
```diff
@@ -574,6 +574,12 @@ export const defaultDeployVpnTunnel = (
     peerGateway: "",
     peerNetworks: [],
     perfectForwardSecrecy: "",
+    state: "unknown",
+    tunnelAttributes: {
+      connections: {},
```
Comment (neurosnap) [inline src/schema/factory.ts:579]: What guarantees can we make about the fields on these objects?  Are they always present from the API?  Can they be null?  Where I'm going with this is we probably want to make guarantees here which means we should list out every field that needs to be available and provide sane defaults when we create this factory.

**Review by neurosnap — COMMENTED** [review]
Overall looks good!  Nice job.  I have a couple of comments.

File: src/schema/factory.ts (hunk 2 above)
Comment (mdelaossa) [reply src/schema/factory.ts:579]: these fields can all be null (if present though they have all the attributes in the type), but it's guaranteed that they're the only three fields that would come in. Should I provide defaults as if these three fields along with their attributes came in, or s97ld I provide the minimal (a fully empty object?)

File: src/deploy/vpn-tunnel/index.ts (hunk 1 above)
Comment (mdelaossa) [reply src/deploy/vpn-tunnel/index.ts]: it's not really a *string* utility, but sure!

**Review by neurosnap — APPROVED** [review]
nice job!

**Pull request merged**